### ISSUE TITLE
[Bug Report] YouTube search fails - Failed to match ytInitialData JSON object

### ISSUE THREAD

**Issue by vastofficial** [jagrosh/MusicBot#520]
**Describe the bug**
Does not query YouTube. Cannot properly connect to YouTube to find search query.

Result in console:
[20:46:14] [WARN] [YoutubeSearchProvider]: Failed to match ytInitialData JSON object
[20:46:26] [WARN] [YoutubeSearchProvider]: Failed to match ytInitialData JSON object

Result in channel:
:bulb: No results found for eminem rap god

**To Reproduce**
Command: !play eminem rap god

**Expected behavior**
Should load the video from the first result on YouTube.

**Debug Info**
```
System Properties:
  java.version = 1.8.0_232
  java.vm.name = OpenJDK 64-Bit Server VM
  java.vm.specification.version = 1.8
  java.runtime.name = OpenJDK Runtime Environment
  java.runtime.version = 1.8.0_232-b09
  java.specification.version = 1.8
  os.arch = amd64
  os.name = Linux

JMusicBot Information:
  Version = 0.2.10
  Owner = 141302663022116864
  Prefix = 1
  AltPrefix = null
  MaxSeconds = 0
  NPImages = false
  SongInStatus = true
  StayInChannel = false
  UseEval = false
  UpdateAlerts = true

Dependency Information:
  JDA Version = 3.8.3_464
  JDA-Utilities Version = 2.1.5
  Lavaplayer Version = 1.3.50

Runtime Information:
  Total Memory = 38
  Used Memory = 26

Discord Information:
  ID = 658610915461169162
  Guilds = 1
  Users = 267
```

**Comment by aljoojOfficial** [comment]
Same probleme here!

**Comment by onlyyouX2** [comment]
Same probleme , And sometimes he out of the room and enters again Is there any solution ?

**Comment by onlyyouX2** [comment]
Same probleme , And sometimes he out of the room and log in again Is there any solution ?

**Comment by vastofficial** [comment]
The issue might be related to YouTube itself. I have been testing other bots and they seemed to have had the same querying issues..

**Comment by onlyyouX2** [comment]
Some other bots It works well.Must be updated MusicBot Version 0.2.11

**Comment by onlyyouX2** [comment]
Some other bots It works well.Must be updated MusicBot Version 0.2.11 .

**Comment by Proxymiity** [comment]
@Sanduhr32 do your PR #501 fixes this bug?

**Comment by brettg0396** [comment]
Having the same issue.

**Comment by Proxymiity** [comment]
I just tested using the fork of LP that 'works', and it seems that it doesn't solve the problem :(

**Comment by Sanduhr32** [comment]
It should work if I update the fork to a newer version.
Tho jag hasn't communicated with me so far.

**Comment by dev-ruby** [comment]
me too..

**Comment by jerise** [comment]
me too..

**Comment by redbrain** [comment]
I have this issue. Does anyone know what the upstream issue is / how it's fixed in forks so that I can patch it in my own Lavaplayer fork?

**Comment by Proxymiity** [comment]
@redbrain The issue is caused by Lavaplayer as it's not actively maintained, and that's what the bot relies on to play music, etc. 
Updating Lavaplayer to Devoxin's fork is the solution, we just need to sit and wait until jagrosh releases a new version of the bot.

**Comment by jagrosh** [comment]
This will be fixed in version 0.3.1

**Comment by yasuobey** [comment]
i am waiting @jagrosh

**Comment by Proxymiity** [comment]
That's great, it'll also close issue #500 👌

**Comment by Arisstath** [comment]
A temporary fix can be found at this precompiled jar https://cdn.discordapp.com/attachments/764904621604601879/769272382640619520/JMusicBot-Snapshot-All.jar

**Comment by MichailiK** [comment]
Keep in mind that jagrosh & others will not provide support for jar files not downloaded from [Releases](https://github.com/jagrosh/MusicBot/releases) of this repo.
Use the above jar file at your own risk.

**Comment by Proxymiity** [comment]
Distributing precompiled jars isn't a potential security flaw?

**Comment by MichailiK** [comment]
> Distributing precompiled jars isn't a potential security flaw?

Yes, which is why I stated to use it at your own risk.

**Comment by jagrosh** [comment]
Please wait for an official release (likely sometime this weekend), as updating lavaplayer is not the only change, and as already stated by others, anybody could create a malicious jar and attempt to pass it off as a copy of this bot.

**Comment by vastofficial** [comment]
As it may seem, YouTube-dl has been taken down by Google: https://twitter.com/EFF/status/1319787243184123904

**Comment by MichailiK** [comment]
Yes. This is not related to MusicBot though, as MusicBot uses Lavaplayer.

**Comment by vastofficial** [comment]
Seems like RIAA had an issue with it 🙄

**Comment by vastofficial** [comment]
@MichailiK Ah okay..

**Comment by Tamim-ul-Alam-Chy** [comment]
i think the problem is with lavaplayer

**Comment by jagrosh** [comment]
This should be resolved as of version 0.3.1
